src/knownnodes.py

```diff
@@ -98,7 +98,7 @@ def saveKnownNodes(dirName=None):
 def addKnownNode(stream, peer, lastseen=None, is_self=False):
     knownNodes[stream][peer] = {
         "lastseen": lastseen or time.time(),
-        "rating": 0,
+        "rating": 1 if is_self else 0,
```

Comment:
I see, so that it doesn't unnecessarily expire / isn't unnecessarily excluded, makes sense.

Reply:
Moreover, I doubt that expired and low rating nodes should be deleted. Because then them may appear again from `addr` messages with rating 0.